aiosmtpd/smtp.py

```diff
@@ -600,7 +610,11 @@ def smtp_DATA(self, arg):
                 data[i] = text[1:]
         content = original_content = EMPTYBYTES.join(data)
         if self._decode_data:
-            content = original_content.decode('utf-8')
+            if self.enable_SMTPUTF8:
+                content = original_content.decode(
+                    'utf-8', errors='surrogateescape')
+            else:
+                content = original_content.decode('ascii', errors='strict')
```

Comment:
Similar to above, this should catch `UnicodeDecodeError` and handle it specially.